vm/systemSmartContracts/staking.go

```diff
@@ -197,20 +213,35 @@ func (r *stakingSC) slash(args *vmcommon.ContractCallInput) vmcommon.ReturnCode
 	}
 
 	var registrationData stakingData
-	data := r.eei.GetStorage(args.Arguments[0].Bytes())
-	err := json.Unmarshal(data, registrationData)
+	stakerAddress := args.Arguments[0].Bytes()
+	data := r.eei.GetStorage(stakerAddress)
+	if data == nil {
+		return vmcommon.UserError
+	}
+
+	err := json.Unmarshal(data, &registrationData)
 	if err != nil {
 		log.Error("unmarshal error on slash function" + err.Error())
 		return vmcommon.UserError
 	}
 
-	if len(data) == 0 {
-		log.Error("slash error: validator was not registered")
+	if !registrationData.Staked {
+		log.Error("cannot slash already unstaked or user not staked")
 		return vmcommon.UserError
 	}
 
-	operation := big.NewInt(0).Set(registrationData.StakeValue)
-	registrationData.StakeValue = registrationData.StakeValue.Sub(operation, args.Arguments[1])
+	stakedValue := big.NewInt(0).Set(registrationData.StakeValue)
+	slashValue := args.Arguments[1]
+	registrationData.StakeValue = registrationData.StakeValue.Sub(stakedValue, slashValue)
+	registrationData.Staked = false
```

Comment:
Should the Staked be set to false after slashing? This means that if we slash just a little of the stake the validator will not be able to validate any longer. 
Please add this test in the TestStakingSC_ExecuteSlashStaked, that the Staked is false for the slashed account.

Also add a test to do two times slashing of 70 with a minimum stake of 100. See what happens. Should the second one return ok or err?

Reply:
After a discussion with Robert, Staked should not change it value on slash. So i remove the  code where Staked is set to false on slash.

Reply:
The second  slash for the moment should not return an error. For the moment slash function know just to subtract the slash value from the staked value and save in storage modifications.